### PR TITLE
{CI} Update bot config

### PR DESCRIPTION
**Pull request by wangzelin007**
**Related command**

**Description**

1. Add a new label `Azure Deployments` for `az deployment\bicep\stack\deployment-scripts\ts\group export` issues 
2. Prevent Merge on "do-not-merge" Label

Test screenshot:
Add "do-not-merge" label
image
image
Remove "do-not-merge" label
image
image

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

**Review by yonzhan — APPROVED** [review]

**Review by anthony-c-martin — APPROVED** [review]
Thank you for making this change!

**Review by bebound — DISMISSED** [review]

File: .github/workflows/BlockPRMerge.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Prevent Merge on "do-not-merge" Label
```
Comment (jsntcy) [inline .github/workflows/BlockPRMerge.yml:1]: Better to name it as `Block PR merging on "do-not-merge" label`

File: .github/workflows/BlockPRMerge.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  block-merge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check "do-not-merge" label
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const labels = context.payload.pull_request.labels.map(label => label.name);
+            if (labels.includes("do-not-merge")) {
+              console.log("PR contains 'do-not-merge' label. Blocking merge.");
```
Comment (jsntcy) [inline .github/workflows/BlockPRMerge.yml:17]: Change message to `It is not allowed to merge a PR with 'do-not-merge' label.`?

File: .github/workflows/BlockPRMerge.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+              process.exit(1); // Exit with a non-zero status code to block merge
+            } else {
+              console.log("PR does not contain 'do-not-merge' label. Allow merge.");
```
Comment (jsntcy) [inline .github/workflows/BlockPRMerge.yml:20]: `It is allowed to merge a PR without 'do-not-merge' label.`

**Review by jsntcy — APPROVED** [review]

**Pull request merged**